### PR TITLE
add a combine_attrs parameter to Dataset.merge

### PR DESCRIPTION
**Pull request by keewis**
This allows passing `combine_attrs` to `Dataset.merge`, which currently defaults to `"override"`.

- [x] Follow-up to #4827
- [x] Tests added
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/core/dataset.py [hunk 1]
```diff
@@ -3934,9 +3935,23 @@ def merge(
             - 'left': use indexes from ``self``
             - 'right': use indexes from ``other``
             - 'exact': error instead of aligning non-equal indexes
+
         fill_value : scalar or dict-like, optional
             Value to use for newly missing values. If a dict-like, maps
             variable names (including coordinates) to fill values.
+        combine_attrs : {"drop", "identical", "no_conflicts", "drop_conflicts", \
+                        "override"}, default: "override"
+            String indicating how to combine attrs of the objects being merged:
```
Comment (dcherian) [inline xarray/core/dataset.py:3944]: I guess this is `Dataset.attrs` not `Dataset[name].attrs`? We should mention that.

**Review by dcherian — APPROVED** [review]

File: xarray/core/dataset.py (hunk 1 above)
Comment (keewis) [reply xarray/core/dataset.py:3944]: I am planning to open another PR that adds support for merging `Dataset[name].attrs` (#4827 already added tests for that), so I think we can keep it as is.

Comment (dcherian) [reply xarray/core/dataset.py:3944]: :+1:  thanks!

File: doc/whats-new.rst [hunk 2]
```diff
@@ -81,11 +81,14 @@ New Features
   contain missing values; 8x faster in our benchmark, and 2x faster than pandas.
   (:pull:`4746`);
   By `Maximilian Roos <https://github.com/max-sixty>`_.
+- Performance improvement when constructing DataArrays. Significantly speeds up repr for Datasets with large number of variables.
 - Add :py:meth:`Dataset.plot.quiver` for quiver plots with :py:class:`Dataset` variables.
   By `Deepak Cherian <https://github.com/dcherian>`_.
 - add ``"drop_conflicts"`` to the strategies supported by the ``combine_attrs`` kwarg
   (:issue:`4749`, :pull:`4827`).
   By `Justus Magin <https://github.com/keewis>`_.
+- allow passing ``combine_attrs`` to :py:meth:`Dataset.merge` (:pull:`4895`).
```
Comment (mathause) [inline doc/whats-new.rst:90]:
```suggestion
- Allow passing ``combine_attrs`` to :py:meth:`Dataset.merge` (:pull:`4895`).
```

File: xarray/core/dataset.py [hunk 3]
```diff
@@ -3934,9 +3935,23 @@ def merge(
...
+
+            - "drop": empty attrs on returned Dataset.
```
Comment (mathause) [inline xarray/core/dataset.py:3946]:
```suggestion
            - "drop": empty attrs on the returned Dataset.
```

**Review by mathause — APPROVED** [review]

File: xarray/core/dataset.py (hunk 3 above)
Comment (keewis) [reply xarray/core/dataset.py:3946]: that was copied from `merge` and should be the same for all functions / methods that accept `combine_attrs` so I would change that in a new PR.

**Comment by keewis** [comment]
thanks for the reviews, @dcherian, @mathause

**Pull request merged**